bibliopixel/led/coord_map.py

```diff
@@ -0,0 +1,143 @@
+from .. util import pointOnCircle
+
+
+class Rotation:
+    ROTATE_0 = 0  # no rotation
+    ROTATE_90 = 3  # rotate 90 degrees
+    ROTATE_180 = 2  # rotate 180 degrees
+    ROTATE_270 = 1  # rotate 270 degrees
+
+
+def rotate_and_flip(coord_map, rotation, flip):
+    for i in range(rotation):
+        coord_map = list(zip(*coord_map[::-1]))
+
+    if flip:
+        coord_map = coord_map[::-1]
+
+    return coord_map
+
+
+def gen_matrix(dx, dy, serpentine=True, offset=0,
+               rotation=Rotation.ROTATE_0, y_flip=False):
+    """Helper method to generate X,Y coordinate maps for strips"""
+
+    result = []
+    for y in range(dy):
+        if not serpentine or y % 2 == 0:
+            result.append([(dx * y) + x + offset for x in range(dx)])
+        else:
+            result.append([((dx * (y + 1)) - 1) - x + offset for x in range(dx)])
+
+    result = rotate_and_flip(result, rotation, y_flip)
+
+    return result
+
+
+def gen_cube(dx, dy, dz, xy_serpentine=True, offset=0,
+             xy_rotation=Rotation.ROTATE_0, z_rotation=Rotation.ROTATE_0,

```

Comment:
Agreed. And yes... this was just to get SimPixel out the door. Normally I would have kept Cube separate, but I needed a 3D object to test. And I'm fine with an updated layout paradigm but just remember that SimPixel will work much better if it takes the pixel layout Once and Only Once. That layout in WebGL is expensive. So as long as it's BiblioPixel that's moving things around, it'll be fine. SimPixel is intentionally made a little dumb...